Sources/SmithyReadWrite/DocumentWritingClosure.swift

```diff
@@ -7,4 +7,12 @@
 
 import struct Foundation.Data
 
-public typealias DocumentWritingClosure<T, Writer> = (T, WritingClosure<T, Writer>) throws -> Data
+public typealias DocumentWritingClosure<T, Writer> = (T, WritingClosure<T, Writer>) throws -> Data?
+
```

Comment:
This helper method for constructing a document writing closure was moved in from `ClientRuntime` and made generic to any `SmithyWriter`.